flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/TableEnvironment.scala

```diff
@@ -99,6 +101,9 @@ abstract class TableEnvironment(val config: TableConfig) {
   // a counter for unique attribute names
   private val attrNameCntr: AtomicInteger = new AtomicInteger(0)
 
+  // registered external catalog names -> catalog
+  private val externalCatalogs = new HashMap[String, ExternalCatalog]
```

Comment:
@fhueske , store catalog aims to get the registered catalog in the future. User may want to call create/update/delete table or database method of catalog when execute DDL in SQL or similar operation in TableAPI.